frontend/src/lib/components/IntervalFilter/intervalFilterLogic.ts

```diff
@@ -33,6 +43,68 @@ export const intervalFilterLogic = kea<intervalFilterLogicType>({
         setFilters: ({ filters }, _, __, previousState) => {
             const { date_from, date_to } = filters
             const previousFilters = selectors.filters(previousState)
+
+            let activeUsersMath: BaseMathType.WeeklyActiveUsers | BaseMathType.MonthlyActiveUsers | null = null
+
+            // We disallow grouping by certain intervals for weekly active users and monthly active users views
+            // e.g. WAUs grouped by month. Here, look for the first event/action running WAUs/MAUs math and
+            // pass that down to the interval filter to determine what groupings are allowed.
+            for (const series of [...(values.filters.events || []), ...(values.filters.actions || [])]) {
+                if (series.math === BaseMathType.WeeklyActiveUsers) {
+                    activeUsersMath = BaseMathType.WeeklyActiveUsers
+                    break
+                }
+
+                if (series.math === BaseMathType.MonthlyActiveUsers) {
+                    activeUsersMath = BaseMathType.MonthlyActiveUsers
+                    break
+                }
+            }
+
+            const enabledIntervals: Intervals = { ...intervals }
+
+            if (activeUsersMath) {
+                // Disallow grouping by hour for WAUs/MAUs as it's an expensive query that produces a view that's not useful for users
+                enabledIntervals.hour = {
```

Comment:
Why not set only `disabledReason`, as opposed to the whole option entry? `label` and `newDateFrom` are not changed from the original as far as I can tell

Reply:
This was something that took me quite a little while to figure out :D 

So what I want each time on filters changing is to get the default values back (reset the state) and check if anything should be disallowed. I need to get the defaults back because the user may have e.g. moved to "total count" from WAUs. 

However, what I have is a shallow copy of `intervals` - i.e. `enabledIntervals` is not the same object as `intervals`, but its internal references to nested objects are!

Thus, if I just change `disabledReason`, I'm changing the `intervals` const. So I need to update the whole object or do a deep copy. I opted for this

Reply:
Ah, of course. But just so that the originals and those copies here don't ever diverge, let's use the spread operator to copy: `...enabledInterval.hour`